### PR TITLE
Transfer ownership function polishing

### PR DESCRIPTION
**Pull request by wiswedel**
Signed-off-by: Sascha Wiswedel <sascha.wiswedel@nextcloud.com>

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Comment by wiswedel** [comment]
The padding after the numbers is quite narrow, but I'm incapable of fixing that. Can someone please have a look at that?

image

I think it's logically safe to push the solution into this polishing PR.

**Comment by ChristophWurst** [comment]
> The padding after the numbers is quite narrow, but I'm incapable of fixing that. Can someone please have a look at that?

This is something that should be changed/fixed broadly with our design rules and not just here so we can have a consistent appearance.

**Comment by wiswedel** [comment]
> > The padding after the numbers is quite narrow, but I'm incapable of fixing that. Can someone please have a look at that?
> 
> This is something that should be changed/fixed broadly with our design rules and not just here so we can have a consistent appearance.

@ChristophWurst Fine for me. How do I get this onto that _design rules fix_ backlog?

File: apps/files/lib/Notification/Notifier.php [hunk 1]
```diff
@@ -122,25 +122,25 @@ public function handleTransferownershipRequest(INotification $notification, stri
...
 						'type' => 'user',
 						'id' => $param['sourceUser'],
 						'name' => $param['sourceUser'],
 					],
 				])
-			->setParsedSubject(str_replace('{user}', $param['sourceUser'], $l->t('Incoming file transfer from {user}')))
+			->setParsedSubject(str_replace('{user}', $param['sourceUser'], $l->t('Incoming ownership transfer from {user}')))
 			->setRichMessage(
-				$l->t('Do you want to accept {path}?'),
+				$l->t('Do you want to accept {path}? Note: The transfer process after accepting may take up to 1 hour.'),
```
Comment (wiswedel) [inline apps/files/lib/Notification/Notifier.php]: @rullzer Beginner's question: How do I add 2 manual line breaks here?

File: apps/files/lib/Notification/Notifier.php [hunk 2]
```diff
@@ -122,25 +122,25 @@ public function handleTransferownershipRequest(INotification $notification, stri
...
 				[
 					'path' => [
 						'type' => 'highlight',
 						'id' => $param['targetUser'] . '::' . $param['nodeName'],
 						'name' => $param['nodeName'],
 					]
 				])
-			->setParsedMessage(str_replace('{path}', $param['nodeName'], $l->t('Do you want to accept {path}?')));
+			->setParsedMessage(str_replace('{path}', $param['nodeName'], $l->t('Do you want to accept {path}? Note: The transfer process after accepting may take up to 1 hour.')));
```
Comment (wiswedel) [inline apps/files/lib/Notification/Notifier.php]: @rullzer same question here about line breaks

**Comment by ChristophWurst** [comment]
> > The padding after the numbers is quite narrow, but I'm incapable of fixing that. Can someone please have a look at that?
> 
> This is something that should be changed/fixed broadly with our design rules and not just here so we can have a consistent appearance.

cc @nextcloud/designers

**Comment by jancborchardt** [comment]
The list styling is a strange way to do this here, so that’s why it looks strange. Nothing wrong with the list styling, instead the flow should be adjusted. Mostly this is about wording:

> **Folder to transfer**
> ( Select )
>
> **New owner**
> [ Search users ]
>
> **( Transfer `foldername` to `username`)**

Transfer button on the bottom should be greyed out until both folder and recipient are selected.

**Comment by wiswedel** [comment]
> ( Transfer `foldername` to `username`)

I'm not capable of implementing this. Anyone else?

**Comment by gary-kim** [comment]
@wiswedel How is this?
Screenshot from 2019-12-15 17-29-00

File: apps/files/lib/Notification/Notifier.php (hunk 1 above)
Comment (rullzer) [reply apps/files/lib/Notification/Notifier.php]: I don't think you can.

Comment (rullzer) [reply apps/files/lib/Notification/Notifier.php]: Ah no you can. Just use `\n` for linebreaks

**Comment by rullzer** [comment]
@gary-kim awesome. Lets go for that

**Comment by gary-kim** [comment]
Made a few changes with @jancborchardt 
@ChristophWurst Do you want to take another look?
Screenshot from 2019-12-16 16-28-21

File: apps/files/lib/Notification/Notifier.php (hunk 1 above)
Comment (wiswedel) [reply apps/files/lib/Notification/Notifier.php]: > Just use \n for linebreaks

That didn't seem to work for me

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
The code changes look good

Touched files like `apps/settings/js/vue-3.js` seem unrelated? Did you install npm dependencies before the build?

**Comment by ChristophWurst** [comment]
Also, this needs a rebase to resolve the conflicts

File: apps/files/lib/Notification/Notifier.php (hunk 1 above)
Comment (juliushaertl) [reply apps/files/lib/Notification/Notifier.php]: `\n` only works inside double quotes.

**Comment by gary-kim** [comment]
Rebased.


> Touched files like `apps/settings/js/vue-3.js` seem unrelated? Did you install npm dependencies before the build?


The CI [webpack-build succeeded](https://drone.nextcloud.com/nextcloud/server/24379/2/3) so it looks like those files should be touched? Let's see if the new build passes.

**Comment by ChristophWurst** [comment]
Yes, let's see. ``+95 −10,994`` is very suspicious, though :thinking:

File: apps/files/lib/Notification/Notifier.php (hunk 1 above)
Comment (wiswedel) [reply apps/files/lib/Notification/Notifier.php]: > `\n` only works inside double quotes.

omg, that's one of those things one has to learn the hard way 🤨

**Comment by gary-kim** [comment]
@ChristophWurst Looks like the [webpack-build is still passing](https://drone.nextcloud.com/nextcloud/server/24418/2/3). I'm also not entirely sure what part of the change resulted in all those removals.

**Review by jancborchardt — APPROVED** [review]
Very nice and simple design-wise now. :) Thanks a lot @gary-kim and @wiswedel!

**Comment by rullzer** [comment]
@ChristophWurst fixed with https://github.com/nextcloud/server/pull/18438

:)

**Review by rullzer — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**